hangman/src/game_state.rs

```diff
@@ -0,0 +1,189 @@
+use std::fmt;
+
+use colored::Colorize;
+
+struct WordSoFar {
+    pub guesses: Vec<char>,
+    pub target_word: String,
+}
+
+impl WordSoFar {
+    pub fn get_remaining_letters(&self) -> i32 {
+        let mut remaining_letters = 0;
+        for char in self.target_word.chars() {
+            if !self.guesses.contains(&char) {
+                remaining_letters += 1;
+            }
+        }
+        remaining_letters
+    }
```

Comment:
This might be able to be simplified with [HashSet::difference](https://doc.rust-lang.org/std/collections/hash_set/struct.HashSet.html#method.difference).